### PR TITLE
Hook up general CPU profiler.

### PR DESCRIPTION
**Pull request by kenzieschmoll**
Clicking "Start recording" will clear the VM's cpu profile buffer. 
Clicking "Stop recording" will pull everything in the buffer and display it in the available profiling views (Call Tree, Bottom Up, Flame Chart).

Confirmed this works with Flutter apps and CLI apps.

Adding documentation for this page here: https://github.com/flutter/website/pull/2835.

### CODE REVIEW

File: packages/devtools/lib/src/performance/performance_controller.dart [hunk 1]
```diff
@@ -3,67 +3,54 @@
...
     _recording = false;
-  }
 
-  void reset() {
-    cpuProfileData = null;
-    recordedProfiles.clear();
-  }
+    // 2^52 is the max int for dart2js. Using this as [extentMicros] for the
+    // getCpuProfile requests will give us all cpu samples we have available.
+    final maxJsInt = pow(2, 52);
```
Comment (jacob314) [inline packages/devtools/lib/src/performance/performance_controller.dart]: nit: make this a top level constant in a utils class and give it type int.

File: packages/devtools/lib/src/profiler/cpu_profile_processor.dart [hunk 2]
```diff
@@ -6,9 +6,9 @@ import 'package:meta/meta.dart';
 import '../utils.dart';
 import 'cpu_profile_model.dart';
 
-/// Protocol for processing [CpuProfileData] and composing it into a structured
-/// tree of [CpuStackFrame]'s.
-class CpuProfileProtocol {
+/// Process for composing [CpuProfileData] into a structured tree of
```
Comment (jacob314) [inline packages/devtools/lib/src/profiler/cpu_profile_processor.dart]: Nit: I know I suggested Processor in the first place but I have a better idea. How about
CpuProfileTransformer.
Then this comment would read better.

File: packages/devtools/lib/src/performance/performance_controller.dart (hunk 1 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/performance/performance_controller.dart]: done.

File: packages/devtools/lib/src/profiler/cpu_profile_processor.dart (hunk 2 above)
Comment (kenzieschmoll) [reply packages/devtools/lib/src/profiler/cpu_profile_processor.dart]: done.

**Pull request merged**